session.go

```diff
@@ -0,0 +1,115 @@
+package mockoidc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// Session is a struct that maintains session state across requests
+type Session struct {
+	SessionID  string
+	Scopes     []string
+	OAuthState string
+	OIDCNonce  string
+	User       *User
+}
+
+// SessionStore is a map of #Session structs
+type SessionStore struct {
+	Store map[string]*Session
+}
+
+// NewSessionStore initializes the SessionStore for this server
+func NewSessionStore() *SessionStore {
+	var ss SessionStore
+	ss.Store = make(map[string]*Session)
+	return &ss
+}
+
+// NewSession returns a pointer to a new Session struct
+func (ss *SessionStore) NewSession(scope string, oAuthState string, oidcNonce string, user *User) (*Session, error) {
+	sessionID, err := nonce(24)
+	if err != nil {
+		return nil, err
+	}
+
+	session := &Session{
+		SessionID:  sessionID,
+		Scopes:     strings.Split(scope, " "),
+		OAuthState: oAuthState,
+		OIDCNonce:  oidcNonce,
+		User:       user,
+	}
+
+	ss.Store[sessionID] = session
+
+	return session, nil
+}
+
+// AccessToken returns the JWT token with the appropriate claims for an access token
+func (s *Session) AccessToken(config *Config, kp *Keypair, now time.Time) (string, error) {
+
+	claims := s.standardClaims(config, config.AccessTTL, now)
+
+	return kp.SignJWT(claims)
+}
+
+// standardClaims returns a populated jwt.StandardCLaims struct
+func (s *Session) standardClaims(config *Config, ttl int, now time.Time) jwt.StandardClaims {
+
+	return jwt.StandardClaims{
```

Comment:
Lets return a reference to this:

```suggestion
func (s *Session) standardClaims(config *Config, ttl int, now time.Time) *jwt.StandardClaims {

	return &jwt.StandardClaims{
```